src/Microsoft.AspNetCore.JsonPatch/Internal/ExpressionHelpers.cs

```diff
@@ -69,21 +69,12 @@ private static string GetPath(Expression expr, bool firstTime)
             }
         }
 
-        private static string GetPropertyNameFromMemberExpression(MemberExpression memberExpression)
+        private static string GetPropertyNameFromMemberExpression(MemberExpression memberExpression, IContractResolver contractResolver)
         {
-            // if there's a JsonProperty attribute, we must return the PropertyName
-            // from the attribute rather than the member name
-            var jsonPropertyAttribute =
-                memberExpression.Member.GetCustomAttribute(
-                typeof(JsonPropertyAttribute), true);
-
-            if (jsonPropertyAttribute == null)
-            {
-                return memberExpression.Member.Name;
-            }
-            // get value
-            var castedAttribute = (JsonPropertyAttribute)jsonPropertyAttribute;
-            return castedAttribute.PropertyName;
+            var jsonObjectContract = (JsonObjectContract)contractResolver.ResolveContract(memberExpression.Expression.Type);
```

Comment:
Are we guaranteed that this will be a valid cast? Could you possibly construct a unit test case where this cast will fail?